server/service/hosts.go

```diff
@@ -2715,3 +2715,61 @@ func (svc *Service) ListHostSoftware(ctx context.Context, hostID uint, opts flee
 	software, meta, err := svc.ds.ListHostSoftware(ctx, host, opts)
 	return software, meta, ctxerr.Wrap(ctx, err, "list host software")
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Host Certificates
+////////////////////////////////////////////////////////////////////////////////
+
+type listHostCertificatesRequest struct {
+	ID uint `url:"id"`
+	fleet.ListOptions
+}
+
+type listHostCertificatesResponse struct {
+	Certificates []*fleet.HostCertificateRecord `json:"certificates"`
+	Meta         *fleet.PaginationMetadata      `json:"meta,omitempty"`
+	Err          error                          `json:"error,omitempty"`
+}
+
+func (r listHostCertificatesResponse) Error() error { return r.Err }
+
+func listHostCertificatesEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (fleet.Errorer, error) {
+	req := request.(*listHostCertificatesRequest)
+	res, meta, err := svc.ListHostCertificates(ctx, req.ID, req.ListOptions)
+	if err != nil {
+		return listHostCertificatesResponse{Err: err}, nil
+	}
+	if res == nil {
+		res = []*fleet.HostCertificateRecord{}
+	}
+	return listHostCertificatesResponse{Certificates: res, Meta: meta}, nil
+}
+
+var listHostCertificatesSortCols = map[string]bool{
+	"common_name":     true,
+	"not_valid_after": true,
+}
+
+func (svc *Service) ListHostCertificates(ctx context.Context, hostID uint, opts fleet.ListOptions) ([]*fleet.HostCertificateRecord, *fleet.PaginationMetadata, error) {
+	if !svc.authz.IsAuthenticatedWith(ctx, authzctx.AuthnDeviceToken) {
+		host, err := svc.ds.HostLite(ctx, hostID)
+		if err != nil {
+			svc.authz.SkipAuthorization(ctx)
+			return nil, nil, ctxerr.Wrap(ctx, err, "failed to load host")
+		}
+		if err := svc.authz.Authorize(ctx, host, fleet.ActionRead); err != nil {
+			return nil, nil, err
+		}
+	}
+
+	// query/after not supported, always include pagination info
+	opts.MatchQuery = ""
+	opts.After = ""
+	opts.IncludeMetadata = true
+	// default sort order is common name ascending
+	if opts.OrderKey == "" || !listHostCertificatesSortCols[opts.OrderKey] {
+		opts.OrderKey = "common_name"
```

Comment:
I opted to silently default to `common_name` if an unsupported column was provided. I don't think we do any validation around that elsewhere, but I can return a 400 if we feel it's better and we want to start being stricter in what we accept.

Reply:
This makes sense based on what we've been doing elsewhere. Do you think it is worth an engineering-initiated story to update our patterns around unsupported URL query params (and potentially get stricter about rejecting unsupported params)?

Reply:
Yeah good idea, I'll create one (the "List hosts" endpoint even crashes when an invalid column is used).

Reply:
I'll address it in a subsequent PR so that Gabe can be unblocked (when you're able to re-approve following the struct change).